docs/python/debugging.md

```diff
@@ -55,11 +55,223 @@ Selecting the **Node.js: Gulp task** yields the following result:
+|**--wait-for-client**     |   none      | **Optional**. Specifies that the code should not run until there's a connection from the debug server. This setting allows you to debug from the first line of your code.        |
+|**--log-to**     |   `<path>`      | **Optional**. Specifies a path to an existing directory for saving logs.         |
+|**--log-to-stderr**     |    none     |  **Optional**. Enables debugpy to write logs directly to stderr.       |
+|**--pid**     |    `<pid>`     | **Optional**. Specifies a process that is already running to inject the debug server into.        |
+|**--configure-<name>** | `<value>` | **Optional**. Sets a debug property that must be known to the debug server before the client connects. Such properties can be used directly in *launch* configuration, but must be set in this manner for *attach* configurations. For example, if you don't want the debug server to automatically inject itself into subprocesses created by the process you're attaching to, use `--configure-subProcess false`.|
+
+> **Note**: `[<arg>]` can be used to pass command line arguments along to the app being launched.
+
+## Debugging by attaching over a network connection
+
+### Local script debugging
+
+In some scenarios, you need to debug a Python script that's invoked locally by another process. For example, you may be debugging a web server that runs different Python scripts for specific processing jobs. In such cases, you need to attach the VS Code debugger to the script once it's been launched:
+
+1. Run VS Code, open the folder or workspace containing the script, and create a `launch.json` for that workspace if one doesn't exist already.
+
+1. In the script code, add the following and save the file:
+
+    ```python
+    import debugpy
+
+    # 5678 is the default attach port in the VS Code debug configurations. Unless a host and port are specified, host defaults to 127.0.0.1
+    debugpy.listen(5678)
+    print("Waiting for debugger attach")
+    debugpy.wait_for_client()
+    breakpoint()
+    ```
+
+1. Open a terminal using **Terminal: Create New Integrated Terminal**, which activates the script's selected environment.
+
+1. In the terminal, install the debugpy package with `python -m pip install --upgrade debugpy`.
+
+1. In the terminal, start Python with the script, for example, `python3 myscript.py`. You should see the "Waiting for debugger attach" message that's included in the code, and the script halts at the `debugpy.wait_for_client()` call.
+
+1. Switch to the Run view, select the appropriate configuration from the debugger drop-down list, and start the debugger.
+
+1. The debugger should stop on the `breakpoint()` call, from which point you can use the debugger normally. You can, of course, set other breakpoints in the script code using the UI instead of using `breakpoint()`.
```

Comment:
`breakpoint()` assumes Python 3.7 or greater. Should we specify that somewhere?

Reply:
@karthiknadig - Is there something different you'd use for versions before Python 3.7? I could add a note or a comment in the code?

Reply:
It is `debugpy.breakpoint()`, this works for all versions. For 3.7 or greater it behaves the same as `breakpoint()`

Reply:
Ah, I see. I updated all the breakpoint() calls to be debugpy.breakpoint() instead. If that works for all versions that makes sense to me to be what we show.

Reply:
On the other hand, we do want the users to use standard Python facilities when available.